rustls/src/client/hs.rs

```diff
@@ -77,21 +77,22 @@ fn find_session(
     sess: &mut ClientSessionImpl,
     dns_name: webpki::DNSNameRef,
 ) -> Option<persist::ClientSessionValue> {
-    let key = persist::ClientSessionKey::session_for_dns_name(dns_name);
-    let key_buf = key.get_encoding();
-
-    let maybe_value = sess
-        .config
-        .session_persistence
-        .get(&key_buf);
+    let key_buf = {
+        let key = persist::ClientSessionKey::session_for_dns_name(dns_name);
+        key.get_encoding()
+    };
 
-    if maybe_value.is_none() {
-        debug!("No cached session for {:?}", dns_name);
-        return None;
-    }
+    let config = &sess.config.session_persistence;
+  
+    let value = match config.get(&key_buf) {
```

Comment:
I would have made this change instead:
```
    let value = sess
        .config
        .session_persistence
        .get(&key_buf)
        .or_else(|| {
            debug!("No cached session for {:?}", dns_name);
            None
        })?;
```

Reply:
Sure, this works as well. I made the change to keep less intermediate variables.